### ISSUE TITLE
GATT CONNECT TIMEOUT(Target device disconnected with status: 8)

### ISSUE THREAD

**Issue by junkchen** [NordicSemiconductor/Android-DFU-Library#295]
**Information**
This GitHub Issue page is for reporting issues or asking questions regarding the iOS DFU library. For general DFU questions, SDK questions, etc, please check our [DevZone](https://devzone.nordicsemi.com/).
Make sure you are using the latest version of the library: [ ![Download](https://api.bintray.com/packages/nordic/android/no.nordicsemi.android%3Adfu/images/download.svg) ](https://bintray.com/nordic/android/no.nordicsemi.android%3Adfu/_latestVersion)
Also, before creating a new issue, make sure similar issue isn't already opened in [open](https://github.com/NordicSemiconductor/Android-DFU-Library/issues) or [closed](https://github.com/NordicSemiconductor/Android-DFU-Library/issues?q=is%3Aissue+is%3Aclosed) issues.

**DFU Bootloader version (please complete the following information):**
 - SDK version: [e.g. SDK 16]
 - Bonding used: no
 - Library version: 1.11.1

**Device information (please complete the following information):**
 - Device: Honor 9,  OPPO Reno
 - OS: Android 9 or Android 10

**Describe the bug**
Hi, When I use Android-DFU-Library to upgrade the firmware, GATT CONNECT TIMEOUT (Target device disconnected with status: 8) often occurs after when device enter to bootloader. When this happens, my phone cannot be connected to the device, only after powering on the device again Before you can connect.
How to solve or avoid this kind of bug? Thanks very much for you!!!

**Logs**
```
2021-03-30 15:23:43.907 28060-28060/ I/DfuBaseService: DFU service created. Version: 1.11.1
2021-03-30 15:23:43.923 28060-28476/ I/DfuBaseService: ---------------------onHandleIntent 开始------------------------
2021-03-30 15:23:43.924 28060-28476/ W/DfuBaseService: Foreground service disabled. Android Oreo or newer may kill a background service few moments after user closes the application.
    Consider enabling foreground service using DfuServiceInitiator#setForeground(boolean)
2021-03-30 15:23:46.972 28060-28476/ I/DfuBaseService: Connecting to the device... 第 1 次
2021-03-30 15:23:46.975 28060-28060/ I/DfuBaseService: connect: dfu 连接设备 Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
2021-03-30 15:23:47.418 28060-28086/ I/DfuBaseService: onConnectionStateChange: DFU 蓝牙已连接
2021-03-30 15:23:47.419 28060-28086/ I/DfuBaseService: Connected to GATT server
2021-03-30 15:23:47.440 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onReceive: ------ ble 蓝牙已连接
2021-03-30 15:23:47.445 28060-28086/ I/DfuBaseService: Attempting to start service discovery... succeed
2021-03-30 15:23:47.450 28060-28060/ I/DfuBaseService: Action received: android.bluetooth.device.action.ACL_CONNECTED
2021-03-30 15:23:47.451 28060-28060/ I/DfuBaseService: onReceive: --------------------- BLE DFU 设备已连接
2021-03-30 15:23:47.945 28060-28076/ I/DfuBaseService: onConnectionUpdated: 连接参数更新：interval=6, latency=0, timeout=500
2021-03-30 15:23:48.287 28060-28076/ I/DfuBaseService: Services discovered
2021-03-30 15:23:48.288 28060-28476/ I/DfuBaseService: 状态参数： mError=0, mConnectionState=-3, mAborted=false
2021-03-30 15:23:48.289 28060-28476/ I/DfuBaseService: onHandleIntent: 获取服务实现 DfuServiceProvider
2021-03-30 15:23:48.302 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onDeviceConnected: ota 升级 设备已连接
2021-03-30 15:23:48.306 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onDfuProcessStarting: ota 升级开始中 deviceAddress=E8:87:65:F8:71:2C
2021-03-30 15:23:48.337 28060-28076/ I/DfuBaseService: onConnectionUpdated: 连接参数更新：interval=39, latency=0, timeout=500
2021-03-30 15:23:48.346 28060-28060/ I/ViewRootImpl: jank_removeInvalidNode all the node in jank list is out of time
2021-03-30 15:23:49.303 28060-28476/ I/DfuImpl: Reading DFU version number...
2021-03-30 15:23:49.411 28060-28476/ I/DfuImpl: Version number read: major.minor=0.1 -> Application with Legacy buttonless update from SDK 7.0 or newer
2021-03-30 15:23:49.414 28060-28476/ I/no.nordicsemi.android.dfu.LegacyButtonlessDfuImpl: isClientCompatible: version=1, assumeDfuMode=false, moreServicesFound=true, servicesSize=4
2021-03-30 15:23:49.414 28060-28476/ I/DfuServiceProvider: getServiceImpl: ----Dfu服务提供器------LegacyButtonlessDfuImpl
2021-03-30 15:23:49.415 28060-28476/ I/DfuImpl: initialize: 进度信息：size=105392, currentPart=1, totalParts=1
2021-03-30 15:23:49.419 28060-28476/ I/DfuImpl: initialize: 设备绑定状态 gatt.getDevice().getBondState()=10
2021-03-30 15:23:49.422 28060-28476/ I/DfuBaseService: onHandleIntent: dfuService 初始化 initialize=true
2021-03-30 15:23:49.422 28060-28476/ W/DfuImpl: Application with legacy buttonless update found
2021-03-30 15:23:49.424 28060-28476/ I/DfuImpl: Enabling notifications...
2021-03-30 15:23:49.434 28060-28476/ I/DfuImpl: enableCCCD: 设置通知结束 cccdEnabled=false, mConnected=true, mError=0, mPaused=false
2021-03-30 15:23:49.434 28060-28476/ I/DfuImpl: enableCCCD: 等待...cccdEnabled=false, mConnected=true, mError=0, mPaused=false
2021-03-30 15:23:49.607 28060-28476/ I/DfuImpl: enableCCCD: wait finish, cccdEnabled=true
2021-03-30 15:23:50.609 28060-28476/ I/DfuImpl: Sending Start DFU command (Op Code = 1, Upload Mode = 4)
2021-03-30 15:23:50.609 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onEnablingDfuMode: ota 升级 已开启dfu模式
2021-03-30 15:23:50.621 28060-28476/ I/DfuImpl: writeOpCode: wait before mRequestCompleted=false, mConnected=true, mError=0, mPaused=false
2021-03-30 15:23:50.925 28060-28476/ I/DfuImpl: writeOpCode: wait after mRequestCompleted=true, mConnected=true, mError=0, mPaused=false
2021-03-30 15:23:50.925 28060-28076/ W/DfuBaseService: Target device disconnected with status: 19
2021-03-30 15:23:50.927 28060-28076/ I/DfuBaseService: onConnectionStateChange: 2-蓝牙已断开
2021-03-30 15:23:50.932 28060-28476/ I/DfuBaseService: Refreshing result: true
2021-03-30 15:23:50.933 28060-28476/ I/DfuBaseService: Cleaning up...   close the GATT.
2021-03-30 15:23:50.943 28060-28476/ I/DfuImpl: Starting service that will connect to the DFU bootloader
2021-03-30 15:23:50.959 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onReceive: ------ ble 蓝牙已断开连接
2021-03-30 15:23:50.961 28060-28060/ I/DfuBaseService: Action received: android.bluetooth.device.action.ACL_DISCONNECTED
2021-03-30 15:23:50.961 28060-28060/ I/DfuBaseService: onReceive: --------------------- BLE DFU 设备连接已断开
2021-03-30 15:23:50.975 28060-28476/ I/DfuBaseService: ---------------------onHandleIntent 结束------------------------
2021-03-30 15:23:50.989 28060-28476/ I/DfuBaseService: ---------------------onHandleIntent 开始------------------------
2021-03-30 15:23:50.990 28060-28476/ W/DfuBaseService: Foreground service disabled. Android Oreo or newer may kill a background service few moments after user closes the application.
    Consider enabling foreground service using DfuServiceInitiator#setForeground(boolean)
2021-03-30 15:23:52.995 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onDeviceConnecting: ota 升级 设备连接中
2021-03-30 15:23:55.998 28060-28476/ I/DfuBaseService: Connecting to the device... 第 2 次
2021-03-30 15:23:56.000 28060-28060/ I/DfuBaseService: connect: dfu 连接设备 Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
2021-03-30 15:23:56.459 28060-28076/ I/DfuBaseService: onConnectionStateChange: DFU 蓝牙已连接
2021-03-30 15:23:56.459 28060-28076/ I/DfuBaseService: Connected to GATT server
2021-03-30 15:23:56.476 28060-28076/ I/DfuBaseService: Attempting to start service discovery... succeed
2021-03-30 15:23:56.494 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onReceive: ------ ble 蓝牙已连接
2021-03-30 15:23:56.495 28060-28060/ I/DfuBaseService: Action received: android.bluetooth.device.action.ACL_CONNECTED
2021-03-30 15:23:56.495 28060-28060/ I/DfuBaseService: onReceive: --------------------- BLE DFU 设备已连接
2021-03-30 15:23:56.983 28060-28086/ I/DfuBaseService: onConnectionUpdated: 连接参数更新：interval=6, latency=0, timeout=500
2021-03-30 15:24:02.035 28060-28086/ W/DfuBaseService: Target device disconnected with status: 8
2021-03-30 15:24:02.035 28060-28086/ I/DfuBaseService: onConnectionStateChange: 2-蓝牙已断开
2021-03-30 15:24:02.036 28060-28476/ I/DfuBaseService: 状态参数： mError=32776, mConnectionState=0, mAborted=false
2021-03-30 15:24:02.036 28060-28476/ I/DfuBaseService: Connection error after: 9041 ms
2021-03-30 15:24:02.036 28060-28476/ E/DfuBaseService: An error occurred while connecting to the device:8
2021-03-30 15:24:02.049 28060-28476/ I/DfuBaseService: Attempt: 1
2021-03-30 15:24:02.063 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onReceive: ------ ble 蓝牙已断开连接
2021-03-30 15:24:02.064 28060-28060/ I/DfuBaseService: Action received: android.bluetooth.device.action.ACL_DISCONNECTED
2021-03-30 15:24:02.065 28060-28060/ I/DfuBaseService: onReceive: --------------------- BLE DFU 设备连接已断开
2021-03-30 15:24:02.066 28060-28476/ I/DfuBaseService: Refreshing result: true
2021-03-30 15:24:02.066 28060-28476/ I/DfuBaseService: Cleaning up...   close the GATT.
2021-03-30 15:24:02.118 28060-28476/ I/DfuBaseService: Restarting the service
2021-03-30 15:24:02.159 28060-28476/ I/DfuBaseService: ---------------------onHandleIntent 开始------------------------
2021-03-30 15:24:02.160 28060-28476/ W/DfuBaseService: Foreground service disabled. Android Oreo or newer may kill a background service few moments after user closes the application.
    Consider enabling foreground service using DfuServiceInitiator#setForeground(boolean)
2021-03-30 15:24:07.170 28060-28476/ I/DfuBaseService: Connecting to the device... 第 3 次
2021-03-30 15:24:07.171 28060-28060/ I/DfuBaseService: connect: dfu 连接设备 Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
2021-03-30 15:24:37.238 28060-28086/ E/DfuBaseService: Connection state change error: 133 newState: 0
2021-03-30 15:24:37.238 28060-28086/ I/DfuBaseService: onConnectionStateChange: 2-蓝牙已断开
2021-03-30 15:24:37.238 28060-28476/ I/DfuBaseService: 状态参数： mError=32901, mConnectionState=0, mAborted=false
2021-03-30 15:24:37.238 28060-28476/ I/DfuBaseService: Connection error after: 33073 ms
2021-03-30 15:24:37.239 28060-28476/ E/DfuBaseService: Device not reachable. Check if the device with address E8:87:65:F8:71:2C is in range, is advertising and is connectable
2021-03-30 15:24:37.239 28060-28476/ I/DfuBaseService: Attempt: 2
2021-03-30 15:24:37.244 28060-28476/ I/DfuBaseService: Refreshing result: true
2021-03-30 15:24:37.244 28060-28476/ I/DfuBaseService: Cleaning up...   close the GATT.
2021-03-30 15:24:37.256 28060-28476/ I/DfuBaseService: Restarting the service
2021-03-30 15:24:37.265 28060-28476/ I/DfuBaseService: ---------------------onHandleIntent 开始------------------------
2021-03-30 15:24:37.266 28060-28476/ W/DfuBaseService: Foreground service disabled. Android Oreo or newer may kill a background service few moments after user closes the application.
    Consider enabling foreground service using DfuServiceInitiator#setForeground(boolean)
2021-03-30 15:24:42.277 28060-28476/ I/DfuBaseService: Connecting to the device... 第 4 次
2021-03-30 15:24:42.279 28060-28060/ I/DfuBaseService: connect: dfu 连接设备 Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
2021-03-30 15:25:12.338 28060-28086/ E/DfuBaseService: Connection state change error: 133 newState: 0
2021-03-30 15:25:12.338 28060-28086/ I/DfuBaseService: onConnectionStateChange: 2-蓝牙已断开
2021-03-30 15:25:12.339 28060-28476/ I/DfuBaseService: 状态参数： mError=32901, mConnectionState=0, mAborted=false
2021-03-30 15:25:12.339 28060-28476/ I/DfuBaseService: Connection error after: 33067 ms
2021-03-30 15:25:12.340 28060-28476/ E/DfuBaseService: Device not reachable. Check if the device with address E8:87:65:F8:71:2C is in range, is advertising and is connectable
2021-03-30 15:25:12.340 28060-28476/ I/DfuBaseService: Attempt: 3
2021-03-30 15:25:12.346 28060-28476/ I/DfuBaseService: Refreshing result: true
2021-03-30 15:25:12.346 28060-28476/ I/DfuBaseService: Cleaning up...   close the GATT.
2021-03-30 15:25:12.963 28060-28060/ I/com.gla.aimattress.module.device.OTAupdataActivity: onDeviceConnected: ota 升级 设备断开了连接
2021-03-30 15:25:12.988 28060-28060/ I/DfuBaseService: DFU service destroyed
```

**Comment by Manmapi** [comment]
Can you pls tell me how you can fix this problem, i got the same error
